src/Data/String/Regex/Unsafe.purs

```diff
@@ -2,13 +2,13 @@ module Data.String.Regex.Unsafe
   ( unsafeRegex
   ) where
 
-import Data.Maybe (fromJust)
-import Data.Either (hush)
+import Prelude (identity)
+import Data.Either (either)
 import Data.String.Regex (Regex, regex)
 import Data.String.Regex.Flags (RegexFlags)
-import Partial.Unsafe (unsafePartial)
+import Partial.Unsafe (unsafeCrashWith)
 
 -- | Constructs a `Regex` from a pattern string and flags. Fails with
 -- | an exception if the pattern contains a syntax error.
 unsafeRegex :: String -> RegexFlags -> Regex
-unsafeRegex s f = unsafePartial fromJust (hush (regex s f))
+unsafeRegex s f = either (unsafeCrashWith identity) (regex s f)
```

Comment:
I think you want `either unsafeCrashWith identity (regex s f)` here. The idea is that we have something equivalent to
```
case regex s f of
  Left err -> unsafeCrashWith err
  Right r -> r
```